pkgs/development/guile-modules/guile-git/default.nix

```diff
@@ -0,0 +1,40 @@
+{ stdenv, fetchurl, guile, libgit2
+, autoreconfHook, pkg-config, texinfo
+}:
+
+let
+  bytestructures = stdenv.mkDerivation rec {
+    pname = "scheme-bytestructures";
+    version = "1.0.7";
+    src = fetchurl {
+      url = "https://github.com/TaylanUB/${pname}/archive/v${version}.tar.gz";
+      sha256 = "1j2jgc0sylxzsizngwlbsjpn663322w48spi1d3zx90akcl04lnb";
+    };
+    nativeBuildInputs = [ autoreconfHook pkg-config ];
+    buildInputs = [ guile ];
+  };
+in stdenv.mkDerivation rec {
+  pname = "guile-git";
+  version = "0.3.0";
+
+  src = fetchurl {
+    url = "https://gitlab.com/${pname}/${pname}/-/archive/v${version}/${pname}-v${version}.tar.gz";
```

Comment:
Is there a reason we can't `fetchFromGitLab`?

---

Nitpick:

It might be just me, but I don't think `pname` should be substituted everywhere -- maybe the last 2 substitutions (`pname` and `version`) are fine, but... meh. I'd rather see something more like `https://gitlab.com/guile-git/guile-git/.....`.

Reply:
See above :p nitpick becomes irrelevant now, but I'll bear it in mind. I had similar feelings.